addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoApi.java

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.net.HttpServiceUtil;
+import org.eclipse.smarthome.core.net.NetUtil;
+import org.openhab.io.neeo.NeeoConstants;
+import org.openhab.io.neeo.NeeoUtil;
+import org.openhab.io.neeo.ServiceContext;
+import org.openhab.io.neeo.internal.models.NeeoAdapterRegistration;
+import org.openhab.io.neeo.internal.models.NeeoRecipe;
+import org.openhab.io.neeo.internal.models.NeeoSystemInfo;
+import org.openhab.io.neeo.internal.net.HttpRequest;
+import org.openhab.io.neeo.internal.net.HttpResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The class provides the API for communicating with a NEEO brain
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class NeeoApi implements AutoCloseable {
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoApi.class);
+
+    /** The gson used in communications */
+    private final Gson gson = NeeoUtil.getGson();
+
+    /** The URL of the brain */
+    private final String brainUrl;
+
+    /** The brain identifier */
+    private final String brainId;
+
+    /** The {@link ServiceContext} to use */
+    private final ServiceContext context;
+
+    /** The {@link HttpRequest} used for making requests */
+    private final HttpRequest request = new HttpRequest();
+
+    /** The known device keys on the brain. */
+    private final NeeoDeviceKeys deviceKeys;
+
+    /** The scheduler used to schedule tasks */
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
I'm not sure if this is allowed? to create your own `Executors` I thought it was discouraged in general, and use supplied schedulers via base framework.

Reply:
Got a hint on how to do that outside of a binding (ie get a scheduler from the framework)?

Reply:
I think this can be done via `private static final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(<String as key for your service>);`

Reply:
Nice!